platform-configuration/src/main/java/org/codice/ddf/configuration/ConfigurationManager.java

```diff
@@ -185,32 +187,72 @@ public ConfigurationManager(List<ConfigurationWatcher> services,
                 .put(HTTP_PORT, getConfigurationValue(PAX_WEB_SERVICE_PID, JETTY_HTTP_PORT));
         readOnlySettings.put(SERVICES_CONTEXT_ROOT,
                 getConfigurationValue(CXF_SERVICE_PID, CXF_SERVLET_CONTEXT));
-        
+
         readOnlySettings.put(KEY_STORE, System.getProperty(SSL_KEYSTORE_JAVA_PROPERTY));
-        readOnlySettings.put(KEY_STORE_PASSWORD,
-                System.getProperty(SSL_KEYSTORE_PASSWORD_JAVA_PROPERTY));
+        readOnlySettings
+                .put(KEY_STORE_PASSWORD, System.getProperty(SSL_KEYSTORE_PASSWORD_JAVA_PROPERTY));
         readOnlySettings.put(TRUST_STORE, System.getProperty(SSL_TRUSTSTORE_JAVA_PROPERTY));
         readOnlySettings.put(TRUST_STORE_PASSWORD,
                 System.getProperty(SSL_TRUSTSTORE_PASSWORD_JAVA_PROPERTY));
-        
 
         this.configuration = new HashMap<String, String>();
 
         // Append the read-only settings to the DDF System Settings so that all
         // settings are pushed to registered listeners
         configuration.putAll(readOnlySettings);
-        
+
         logger.debug("EXITING: ctor");
     }
 
+    public void setProtocol(String protocol) {
+        configurationProperties.put(PROTOCOL, protocol);
+        logger.debug("protocol set to {}", protocol);
+    }
+
+    public void setHost(String host) {
+        configurationProperties.put(HOST, host);
+        logger.debug("host set to {}", host);
+    }
+
+    public void setPort(String port) {
+        configurationProperties.put(PORT, port);
+        logger.debug("port set to {}", port);
+    }
+
+    public void setId(String id) {
+        configurationProperties.put(SITE_NAME, id);
+        logger.debug("site name set to {}", id);
+    }
+
+    public void setVersion(String version) {
+        configurationProperties.put(VERSION, version);
+        logger.debug("version set to {}", version);
+    }
+
+    public void setOrganization(String organization) {
+        configurationProperties.put(ORGANIZATION, organization);
+        logger.debug("organization set to {}", organization);
+    }
+
+    public void setContact(String contact) {
+        configurationProperties.put(CONTACT, contact);
+        logger.debug("contact set to {}", contact);
+    }
+
+    /**
+     * Called once after all managed property setters have been called.
+     */
+    public void init() {
+        updated(Collections.unmodifiableMap(configurationProperties));
+    }
+
     /**
      * Invoked when the DDF system settings are changed in the Admin Console,
      * this method then pushes those DDF system settings to each of the
      * registered ConfigurationWatchers.
-     * 
-     * @param updatedConfig
-     *            list of DDF system settings, not including the read-only
-     *            settings
+     *
+     * @param updatedConfig list of DDF system settings, not including the read-only
```

Comment:
Can you replace list with map and mention that a null reference is acceptable and will be ignored?